apps/web/hooks/useFeatureFlag.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from 'react'
+import { getFeatureFlag } from '../utils/featureFlag'
+
+export const useFeatureFlag = (flag: string, defaultValue: boolean) => {
+  const [state, setState] = useState({
+    loading: true,
+    value: defaultValue,
+  })
+
+  useEffect(() => {
+    let isMounted = true
+    const fetchFlag = async () => {
+      const value = await getFeatureFlag(flag, defaultValue)
+      if (isMounted) setState({ loading: false, value })
+    }
+    fetchFlag()
+
+    return () => (isMounted = false)
+  }, [defaultValue, flag])
+
+  return state
+}
+
+export default useFeatureFlag
```

Comment:
Why not use `libs/react/feature-flags/src/lib/useFeatureFlag.ts`

Reply:
Because the web project does not have the context provider that hook requires. So my approach was to create my own, I was in contact with Eiríkur on slack and he helped me figure out how to add the getFeatureFlag function. https://islandis.slack.com/archives/C019VN1KG04/p1650383631008049